lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -81,23 +81,22 @@ def render_template_with_datadog(*args)
     end
 
     def patch_partial_renderer(klass)
-      # rubocop:disable Metrics/BlockLength
       klass.class_eval do
         def render_with_datadog(*args, &block)
           # Create a tracing context and start the rendering span
           tracing_context = {}
           Datadog::Contrib::Rails::ActionView.start_render_partial(tracing_context: tracing_context)
-          add_tracing_context(tracing_context)
+          tracing_contexts[current_span_id] = tracing_context unless current_span_id.nil?
```

Comment:
I think not having the `unless` conditional here would be better.

* We won't have a leak because we delete the entry anyways;
* If `current_span_id` is `nil` with the current implementation, you'll crash because you're doing hash operations in the following lines without the conditional;

So my understanding is: you either remove it or replicate in all lines referencing `tracing_contexts`

Reply:
Should be addressed now.